tox.ini

```diff
@@ -7,8 +7,7 @@ max-line-length = 100
 [tox]
 skipsdist = True
 skip_missing_interpreters = True
-envlist = fmt, lint, unit, integration, {katib-controller}-{lint,unit,integration}
-
+envlist = fmt, lint, unit, integration, {katib-controller,katib-db-manager}-{lint,unit,integration}
```

Comment:
Not really an issue with the PR, since this already existed, just noting that the way we currently define this `envlist`, tox will also attempt to run stuff like this using the generative testenv:

```
unit: commands[0]> tox -c charms/katib- -e unit
ROOT: HandledError| could not recognize config file charms/katib-
unit: exit 254 (0.13 seconds) /home/phoevos/canonical/kubeflow/katib> tox -c charms/katib- -e unit pid=264201
unit: FAIL ✖ in 0.13 seconds
integration: commands[0]> tox -c charms/katib- -e integration
ROOT: HandledError| could not recognize config file charms/katib-
integration: exit 254 (0.13 seconds) /home/phoevos/canonical/kubeflow/katib> tox -c charms/katib- -e integration pid=264216
integration: FAIL ✖ in 0.19 seconds
```

This is due to the fact that `tox` performs pattern matching using the values from the `envlist` as input to conditionally set these variables:
https://github.com/canonical/katib-operators/blob/885f7334002f4c49675a0e2665d4809fee4f55ae/tox.ini#L20-L22

In the cases of the plain `fmt`, `lint`, `unit`, and `integration` envs included in the `envlist`, `CHARM` is left empty, which leads to these weird tox invocations.

Does this make sense? Noting it here, but this should be moved to an issue.

Reply:
it makes sense, i'll create a separate issue and discuss it there. thanks @phoevos

Reply:
moved to #110